### PR TITLE
Generate String methods for enums with stringer

### PR DESCRIPTION
**Pull request by jakebailey**
Using the oft used: https://cs.opensource.google/go/x/tools/+/refs/tags/v0.26.0:cmd/stringer/stringer.go

This has some rough edges:

- Bitset enums are not handled
	- This seems fixable upstream with another flag.
- `-trimprefix` can't be a list, so we can't generate a single file for all enums and also use that flag.
	- This seems fixable upstream; commas can't have been a legal `trimprefix` before.
- The "first" value of duplicate values always wins.
	- This seems desirable?
- Many calls to `go run` in `go generate` is slow because there's no binary caching, which will be fixed with `go tool stringer` hopefully in an upcoming version

### CODE REVIEW

**Comment by gabritto** [comment]
Are we moving forward with this? I would like to use something in this PR.

**Comment by jakebailey** [comment]
I'd like to, though I'm not sure if people will be happy with one extra file per type to add `String()` to. I'll probably update this PR and bring them all back into a single file again and eventually report the above upstream.

**Comment by jakebailey** [comment]
Merged into one file, so this is ready for review if you'd like to get this in!

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+//go:build tools
+
+package tools
+
+// Until https://github.com/golang/go/issues/48429
+
+import (
+	_ "golang.org/x/tools/cmd/stringer"
+)
```
Comment (jakebailey) [inline tools.go]: This file is a way to "reference" dependencies so they'll appear in `go.mod`, which `go run` will then respect.

It's getting replaced upstream with `go get -tool golang.org/x/tools/cmd/stringer` and `go tool golang.org/x/tools/cmd/stringer`/`go tool stringer` but that's not yet merged in its entirety.

Comment (jakebailey) [reply tools.go]: I'm actually going to move this file into `internal` somewhere to make the error you get when you run `go build` at the root be more reasonable.

Comment (jakebailey) [reply tools.go]: Right now it's:

```
package github.com/microsoft/typescript-go: build constraints exclude all Go files in /home/jabaile/work/TypeScript-go
```

**Pull request merged**